streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableReduce.java

```diff
@@ -89,11 +89,6 @@ public void process(K key, Change<V> value) {
             // update the store with the new value
             store.put(key, newAgg);
 
-            // send the old / new pair
```

Comment:
Nice catch - obviously we don't have any tests for this otherwise it would have been caught out.